tests/test_sphinx_extension.py

```diff
@@ -0,0 +1,124 @@
+in all copies or substantial portions of the Software.
+"""
+
+import os
+import unittest
+from sphinx.application import Sphinx
+from sphinx.util.docutils import docutils_namespace
+from sphinx.errors import SphinxWarning
+from functools import wraps
+
+from tests import tmpdir
+
+
+class MockApp(Sphinx):
+    """
+    A subclass of :class:`Sphinx`
+    """
+
+    def __init__(self,):
+
+        Sphinx.__init__(
+            self,
+            srcdir=os.path.abspath(""),
+            confdir=os.path.abspath(""),
+            outdir="result",
+            doctreedir="doctreedir",
+            buildername="html",
+            warningiserror=True,
+        )
+
+
+class with_sphinx_app(object):
+    def __init__(self, *sphinxargs, **sphinxkwargs):
+        self.sphinxargs = sphinxargs
+        self.sphinxkwargs = sphinxkwargs
+
+    def __call__(self, func):
+        @wraps(func)
+        def decorator(*args, **kwargs):
+            app = None
+            sphinxkwargs = dict(self.sphinxkwargs)
+            with docutils_namespace():
+                app = MockApp(*self.sphinxargs, **sphinxkwargs)
+                return func(*(args + (app,)), **kwargs)
+
+        return decorator
+
+
+class TestSphinxExtension(unittest.TestCase):
+    @tmpdir("tests/data/doc")
+    @with_sphinx_app()
+    def test_invalid_setup(self, app):
+        missing_func = """
+        .. configsuite::
+            :module: tests
+        """
+
+        with open("index.rst", "w") as f:
+            f.write(missing_func)
+
+        with self.assertRaises(SphinxWarning) as context:
+            app.build()
+
+        self.assertTrue(
+            "Expected 'module.function', got tests" in str(context.exception)
+        )
+
+    @tmpdir("tests/data/doc")
+    @with_sphinx_app()
+    def test_invalid_module(self, app):
+        invalid_func = """
+        .. configsuite::
+            :module: tests.non_existing_module.non_existing_function
+        """
+
+        with open("index.rst", "w") as f:
+            f.write(invalid_func)
+
+        with self.assertRaises(SphinxWarning) as context:
+            app.build()
+
+        self.assertTrue("non_existing_module" in str(context.exception))
+
+    @tmpdir("tests/data/doc")
+    @with_sphinx_app()
+    def test_invalid_func(self, app):
+        invalid_func = """
+        .. configsuite::
+            :module: tests.data.non_existing_function
+        """
+
+        with open("index.rst", "w") as f:
+            f.write(invalid_func)
+
+        with self.assertRaises(SphinxWarning) as context:
+            app.build()
+
+        self.assertTrue(
+            "has no attribute 'non_existing_function'" in str(context.exception)
+        )
+
+    @tmpdir("tests/data/doc")
+    @with_sphinx_app()
+    def test_sphinx_build(self, app):
+        app.build()
+        with open(os.path.join(app.outdir, "index.html"), "r") as f:
+            html = f.read()
+
+        self.assertTrue("PetMeet" in html)
```

Comment:
This really isn't much of a test. I thought about adding the html output and test against that instead, but then if anything changes in the builder and sphinx, we would need to update the test - we shouldn't depend on external libraries..

I'll see if it's possible to test the nodes added by the extension instead.

Reply:
Expanded slightly, but did not manage to test the nodes directly